app/src/androidTest/java/com/sdp/movemeet/map/MainMapFragmentTest.java

```diff
@@ -0,0 +1,139 @@
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.sdp.movemeet.LoginActivity;
+import com.sdp.movemeet.MainActivity;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.UploadActivityActivity;
+
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(AndroidJUnit4.class)
+public class MainMapFragmentTest {
+
+    private UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+    private FirebaseAuth fAuth;
+    private FirebaseUser user;
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
+
+   @Rule
+   public FragmentTestRule<?, MainMapFragment> fragmentTestRule =
+           FragmentTestRule.create(MainMapFragment.class);
+
+
+    @Before
+    public void setUp() throws InterruptedException {
+        fAuth = FirebaseAuth.getInstance();
+        fAuth.signInWithEmailAndPassword("victor.carles@epfl.ch", "1234567890").addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task) {
+                user = fAuth.getCurrentUser();
+            }
+        });
+        waitFor(2000);
+    }
+
+
+    @Test
+    public void mainMapFragment_MarkerOnMapForUser() throws UiObjectNotFoundException {
+        UiObject marker = uiDevice.findObject(new UiSelector().descriptionContains("I am here !"));
+        assertNotNull(marker);
+    }
+
+
+    @Test
+    public void mainMapFragment_isDisplayed() throws InterruptedException {
+        onView(withId(R.id.fragment_map)).check(matches((isDisplayed())));
+    }
+
+    @Test
+    public void mainMapFragment_userClickingOnMapAddsNewActivity() throws UiObjectNotFoundException, InterruptedException {
```

Comment:
The problem I have with this method is that, for some reason, the UIAutomator doesn't click on the map, although it should. At first, I thought it was because the user was not logged in (hence not being able to add an activity by clicking on the map), but I first check that he is logged with the assertNotNull(user), so the problem comes from the UIdevice. Do you think I could find a way to test the onMapClick with Mockito ? I've never used it, maybe I should look deeper on it.

Reply:
Ohhh, I see. Mockito could still be used to inject your own test logic in a callback, but I don't think it could help with clicking. Maybe there's some way to make the map focused so that all clicks and interactions go through the map first?